spiketoolkit/sorters/basesorter.py

```diff
@@ -133,14 +135,18 @@ def get_result(self):
             # reassemble the sorting outputs
             sorting_list = [sort for sort in sorting_list if sort is not None]
             multi_sorting = se.MultiSortingExtractor(sortings=sorting_list)
-            return multi_sorting
+            sorting = multi_sorting
+
+        if self.delete_folders:
+            for out in self.output_folders:
+                if self.debug:
+                    print("Removing ", str(out))
+                shutil.rmtree(str(out), ignore_errors=True)
```

Comment:
I already thought about this
I was more for another behavior: remove the folder in **_setup_recording** if it exists. The kargs woulmd **reset_folder_if_exists**.
So a clean before rather than a clean after.

Pros **reset_folder_if_exists** (=delete_folders_before_sort)
  * If a sorter do not finish properlly, you want to re-run, so a remove at the beginning.
  * I want want to run many sorter in parralell and really want to keep the folder to open it in a futur python session.
  * for long task list (multi sorter or multi recording) you can inspect folders present or not.

Pros for **delete_folders** (=delete_folders_after_sort):
  * when it is finish, the disk space is left.

I was also planning to to add this a classmethod ```def get_result_from_folder```.
It would be the same as ```get_result``` but relying only on the disk and not on self.output_folder.
Of course this would not be implement for ms4 for now.

Reply:
I think that **reset_folder_if_exists** and **delete_folders_after_sort** could live together. The way we use it, we run spike sorting and then save the output in a different format (exdir), so we want to delete the folder after sorting. I think this option should be left to the user. 

Do you agree?